packages/toolpad-app/src/server/localMode.ts

```diff
@@ -760,9 +766,17 @@ async function writeThemeFile(root: string, theme: Theme | null) {
   }
 }
 
-async function writeDomToDisk(dom: appDom.AppDom): Promise<void> {
+async function writeDomToDisk(dom: appDom.AppDom, oldDom?: appDom.AppDom): Promise<void> {
```

Comment:
Is it be possible to do this the way `deletePage` works? Where we call `deletePage` from the frontend and don't rely on `writeDomToDisk` to reconcile everything? 

The path forward for the `AppDom` will probably be to split it up in its separate pieces and not read/write it as a whole.